packages/clay-core/src/icon-selector/DropdownSelector.tsx

```diff
@@ -0,0 +1,258 @@
+/**
+ * SPDX-FileCopyrightText: © 2025 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import Button from '@clayui/button';
+const spritemap = require('@clayui/css/lib/images/icons/icons.svg');
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import {
+	InternalDispatch,
+	Overlay,
+	useControlledState,
+	useId,
+	useOverlayPosition,
+} from '@clayui/shared';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+
+export type IProps<T> = {
+	/**
+	 * Flag to indicate if menu is showing or not.
+	 */
+	active?: boolean;
+
+	/**
+	 * Custom button component.
+	 */
+	as?:
+		| 'button'
+		| React.ForwardRefExoticComponent<any>
+		| ((props: React.ComponentProps<typeof Button>) => JSX.Element);
+
+	/**
+	 * Value to display the trigger with an icon or text.
+	 */
+	buttonWithIcon?: boolean;
+
+	/**
+	 * The initial value of the active state (uncontrolled).
+	 */
+	defaultActive?: boolean;
+
+	/**
+	 * Property to set the initial value of `items` (uncontrolled).
+	 */
+
+	defaultItems?: Array<T> | null;
+
+	/**
+	 * The initial value of the input (uncontrolled).
+	 */
+	defaultValue?: string;
+
+	/**
+	 * Direction the menu will render relative to the Icon Selector.
+	 */
+	direction?: 'bottom' | 'top';
+
+	/**
+	 * Property to render content with dynamic data.
+	 */
+	items?: Array<T> | null;
+
+	/**
+	 * Funtion to select an icon
+	 */
+	handleIconSelect: (icon: string) => void;
+
+	/**
+	 * Callback for when the active state changes (controlled).
+	 */
+	onActiveChange?: InternalDispatch<boolean>;
+
+	[key: string]: any;
+};
+
+export default function IconSelector<
+	T extends Record<string, any> | string | number
+>({
+	active: externalActive,
+	as: As = Button,
+	buttonWithIcon = false,
+	children,
+	containerElementRef,
+	defaultActive,
+	direction = 'bottom',
+	handleIconSelect,
+	onActiveChange,
+}: IProps<T>) {
+	const menuRef = useRef<HTMLDivElement>(null);
+	const triggerRef = useRef<HTMLButtonElement | null>(null);
+
+	const ariaControls = useId();
+
+	const [iconNames, setIconNames] = useState<Array<string>>([]);
+	const [searchTerm, setSearchTerm] = useState<string>('');
+
+	const [active, setActive] = useControlledState({
+		defaultName: 'defaultActive',
+		defaultValue: defaultActive,
+		handleName: 'onActiveChange',
+		name: 'active',
+		onChange: onActiveChange,
+		value: externalActive,
+	});
+
+	const filteredIcons = iconNames.filter((icon) =>
+		icon.toLowerCase().includes(searchTerm.toLowerCase())
+	);
+	const onClose = useCallback(() => setActive(false), []);
+
+	useOverlayPosition(
+		{
+			alignmentByViewport: true,
+			alignmentPosition: direction === 'bottom' ? 5 : 7,
```

Comment:
Ask Matu if we can make these exported constants from Overlay so it's more readable.

Reply:
Hey @ethib137 , to export this constants, there should have a major refactor on the useOverlayPosition and the alignment for dropdowns . I think  is better if we keep an `enum` inside the icon selector and work on this refactor in another story. What do you think?
cc: @matuzalemsteles

Reply:
That's fine with me as long as we create a ticket now for that refactor.

Reply:
@ilzamcmed can you create a ticket for that and share it here?

Reply:
Sure! He it is https://liferay.atlassian.net/browse/LPD-48999